src/account/AccountPickScreen.js

```diff
@@ -17,15 +19,20 @@ type Props = $ReadOnly<{|
 |}>;
 
 class AccountPickScreen extends PureComponent<Props> {
-  handleAccountSelect = (index: number) => {
+  handleAccountSelect = async (index: number) => {
     const { accounts, dispatch } = this.props;
     const { realm, isLoggedIn } = accounts[index];
     if (isLoggedIn) {
       setTimeout(() => {
         dispatch(switchAccount(index));
       });
     } else {
-      dispatch(navigateToRealmScreen(realm));
+      try {
+        const serverSettings: ApiResponseServerSettings = await api.getServerSettings(realm);
+        dispatch(navigateToAuth(serverSettings));
+      } catch (err) {
+        Alert.alert('Cannot connect to server', '', [{ text: 'OK' }]);
+      }
```

Comment:
@ray-kraesig 
Thanks for your time,
I am working on it now

Reply:
can you give an idea on how to hide the button for second click?
 if displaying loadingIndicator, where can i put it?


> * An unguarded `await` in a UI handler makes a race condition possible. If the user taps on the login button, gets impatient while a connection is being made, and taps the button again, the UI will still be responsive to the second tap, and will try to start another login sequence.
>   (This is handled in `RealmScreen.js` with the `progress` entry in the component state, which simultaneously shows a busy-indicator and disables the relevant button. Some analogous construct seems to be needed here.)

Reply:
My first impulse would be a translucent light-grey overlay with an opaque spinner on it. I'm not sure if that's what we really want here, but it would be a reasonable first cut. @gnprice?

Of note, however:

* You can't just hide the single button the user clicked; you have to hide – or disable – _all_ the buttons, for the same reason. (Hence the suggestion of a translucent overlay.)
* The user must be able to cancel out of the action with the standard back button/gesture (on Android) and/or a "Cancel" button (on iOS).
  * Doing this properly will require react-native v0.60, since that was when `AbortController` [was introduced](https://github.com/react-native-community/releases/blob/master/CHANGELOG.md#060); but a version which ignores the fetch result rather than actually cancelling should still be feasible.

Reply:
(Replied in the main thread for a bit more room, as I found I'd written a fairly long and detailed reply.)